cpp/include/cudf/detail/gather.cuh

```diff
@@ -483,7 +483,7 @@ struct column_gatherer_impl<struct_view> {
     auto const nullable =
       nullify_out_of_bounds || std::any_of(sliced_children.begin(),
                                            sliced_children.end(),
-                                           [](auto const& col) { return col.nullable(); });
+                                           [](auto const& col) { return col.has_nulls(); });
```

Comment:
Quite a few of the changes in this PR (here and below) don't belong in a PR aimed at introducing new example code. They should be fixed in a separate PR.

Reply:
So if this PR requires such fix to work, please post a fix PR and mark as dependency of this one.

Reply:
suggestion (non-blocking): Ideally I'd also split the nvtx range changes from the nullable->has_nulls changes since they're fairly different classes of issues.

Reply:
Looks like this was done in a sequence of commits (e.g. https://github.com/rapidsai/cudf/pull/13795/commits/f2c42be2c8868a23375892921f7f9aae0544568b for range).